### DISCUSSION TITLE
Should we use provider URI (like host[/path?]) instead of https URL?

### DISCUSSION THREAD

**Discussion by gazarenkov** [redhat-appstudio/service-provider-integration-operator#498] [Ideas]
It seems the protocol is not needed yet adds some difficulties.

**Comment by sparkoo** [comment]
I think we need to clarify here to what exactly this apply. Is it user input? SPI configuration? Anything else?

I think for user input (`repositoryUrl` in the Binding and `serviceProviderUrl` in the Token) we can support both, with and without https. I would say for code repo it's typical for user to copy-paste from browser url so it is typically with https. For docker images, I would say it's mostly without https asi I would imagine user copy-paste that from image name tag.
We kind of agreed that if there is protocol, it must be https, if there is not, we assume https.

Question is, whether we want to update CR spec to always have scheme? For example user creates a binding with repourl `quay.io/redhat-appstudio/spi-operator`, should we update spec to `https://quay.io/redhat-appstudio/spi-operator`?

**Reply by metlos** [reply]
IMHO we should never overwrite the spec. If we want to be explicit, we can add some fields to the status (like `effectiveServiceProviderUrl` or some such) but we should leave the spec as specified by the user.

Otherwise I agree with your reasoning for assuming `https` if not specified.

**Reply by sparkoo** [reply]
nice, this thing can do replies ... :))

I'm afraid we're actually doing that already https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/controllers/spiaccesstokenbinding_controller.go#L624
